### PR TITLE
fix: postcss composes directive

### LINKED ISSUE #2385: Investigate CSS composition
We have a need to avoid superflous classes so in an effort we want to investigate how we can combine classes into one class. 

- "compose" paragraph style in button
- Test post-css plugin for mixins to get merged css for composition
- or just change how we define font related properties in our components

### PR DESCRIPTION
**Pull request by eirikbacker**
- Closes #2385
- Using same syntax as postcss `composes`, but custom implementation for `postcss`

### CODE REVIEW

File: packages/css/postcss.config.js [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
     require('autoprefixer'),
   ],
 };
+
+function postcssApply() {
+  return {
+    postcssPlugin: '@apply', // Add support for @apply directive to enable inlining utility classes
+    AtRule: {
+      apply: (rule) => {
+        rule.root().walkRules((from) => {
```
Comment (mimarz) [inline packages/css/postcss.config.js:18]: I guess this assumes referenced rules are imported before and thus available? Good to know if we want to use this for other use-cases.

**Comment by mimarz** [comment]
How do we want to handle multiple applies? 

1. ```css
   @apply ds-focus, ds-paragraph--md;
   ```
2. ```css 
   @apply ds-focus ds-paragraph--md;
   ```
3. ```css 
   @apply ds-focus 
   @apply ds-paragraph--md;
   ```

**Comment by mimarz** [comment]
Do we have to fix the build for individual files? When the rule is not found it just keeps the `@apply ds-focus`.

File: packages/css/postcss.config.js (hunk 1 above)
Comment (eirikbacker) [reply packages/css/postcss.config.js:18]: Yes, just like tailwind, a rule needs to exist in the same code base, but import order does not matter :)

**Comment by eirikbacker** [comment]
> Do we have to fix the build for individual files? When the rule is not found it just keeps the `@apply ds-focus`.

The individual files are not processed at all by postcss as you can see, so I'm a bit unsure why we add them to dist?

**Comment by eirikbacker** [comment]
> How do we want to handle multiple applies?
> 
> 1. ```css
>    @apply ds-focus, ds-paragraph--md;
>    ```
> 2. ```css
>    @apply ds-focus ds-paragraph--md;
>    ```
> 3. ```css
>    @apply ds-focus 
>    @apply ds-paragraph--md;
>    ```

I suggest the #3 approach - no problem making the others work, but it will require more code in your little plugin, and one apply class per line seems a bit cleaner too? ☺️

**Comment by mimarz** [comment]
After a short meeting between me and @eirikbacker we decided to tweak the `@apply` to a `@composes` with path to css file like css modules. ieg; @composes ds-focus from './utilities.css';

This will make it more clear and also work for single component css files having it applied correctly.

**Review by mimarz — APPROVED** [review]
Make sure to update the description and pr title with new directive name :)

**Comment by unekinn** [comment]
Bra jobba! 👏

**Pull request merged**